### DISCUSSION TITLE
How to hide or change url?

### DISCUSSION THREAD

**Discussion by messixukejia** [vercel/next.js#39155] [App Router]
How to hide or change url?

**Comment by messixukejia** [comment]
a

**Comment by wiggledom** [comment]
I guess there are two ways to hide the url in react

1. Using Memory History
Memory history doesn't manipulate or read from the address bar. This is how we implement server rendering. It's also useful for testing and other rendering environments (like React Native).

Use one like so:

const history = createMemoryHistory(location)
render(
  <Router history={history}>
    <Route path='/' component={App}>
      <IndexRoute component={Home} />
      <Route path='about' component={About} />
      <Route path='features' component={Features} />
    </Route>
  </Router>,
  document.getElementById('app')
)

2. Using MemoryRouter

A Router that keeps the history of your "URL" in memory (does not read or write to the address bar). Useful in tests and non-browser environments like React Native.

import { MemoryRouter } from 'react-router'

<MemoryRouter>
  <App/>
</MemoryRouter>


I 'd appreciate it if these soutions can help you
Thanks
